Student/StudentUnitTests/Assignments/AssignmentDetails/AssignmentDetailsPresenterTests.swift

```diff
@@ -190,6 +190,174 @@ class AssignmentDetailsPresenterTests: PersistenceTestCase {
+        wait(for: [expectation], timeout: 1)
+        XCTAssertTrue( presenter.submissionTypesSectionIsHidden() )
+    }
+
+    func testSubmissionTypesSectionNotHiddenAfterAvailability() {
+        setupIsHiddenTest(lockStatus: .after)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.submissionTypesSectionIsHidden() )
+    }
+
+    func testSubmissionTypesSectionNotHidden() {
+        setupIsHiddenTest(lockStatus: .unlocked)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.submissionTypesSectionIsHidden() )
+    }
+
+    func testGradesSectionIsHiddenBeforeAvailability() {
+        setupIsHiddenTest(lockStatus: .before)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertTrue( presenter.gradesSectionIsHidden() )
+    }
+
+    func testGradesSectionNotHiddenAfterAvailability() {
+        Course.make()
+        Assignment.make(from: .make(submission: APISubmission.make(workflow_state: .graded), unlock_at: Date().addYears(-1), locked_for_user: true, lock_explanation: "this is locked"))
+        presenter.viewIsReady()
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.gradesSectionIsHidden() )
+    }
+
+    func testGradesSectionNotHidden() {
+        Course.make()
+        Assignment.make(from: .make(submission: APISubmission.make(workflow_state: .graded)))
+        presenter.viewIsReady()
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.gradesSectionIsHidden() )
+    }
+
+    func testViewSubmissionButtonSectionIsHiddenBeforeAvailability() {
+        setupIsHiddenTest(lockStatus: .before)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertTrue( presenter.viewSubmissionButtonSectionIsHidden() )
+    }
+
+    func testViewSubmissionButtonSectionNotHiddenAfterAvailability() {
+        setupIsHiddenTest(lockStatus: .after)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.viewSubmissionButtonSectionIsHidden() )
+    }
+
+    func testViewSubmissionButtonSectionNotHidden() {
+        setupIsHiddenTest(lockStatus: .unlocked)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.viewSubmissionButtonSectionIsHidden() )
+    }
+
+    func testDescriptionSectionIsHiddenBeforeAvailability() {
+        setupIsHiddenTest(lockStatus: .before)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertTrue( presenter.descriptionIsHidden() )
+    }
+
+    func testDescriptionSectionNotHiddenAfterAvailability() {
+        setupIsHiddenTest(lockStatus: .after)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.descriptionIsHidden() )
+    }
+
+    func testDescriptionSectionNotHidden() {
+        setupIsHiddenTest(lockStatus: .unlocked)
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse( presenter.descriptionIsHidden() )
+    }
+
+    func testsubmitAssignmentButtonIsHiddenBeforeAvailability() {
```

Comment:
These last 3 tests need capital `S` to get picked up by the test runner. `testSubmit...`

Reply:
doh